### PR TITLE
[MRG] Text Regressor using BERT Model

### LINKED ISSUE #594: Text Regressor
### Feature Description

Provide Text Regressor module.

### Reason

Provide Text Regressor.

### Solution

Use Google AI's BERT model.

### PR DESCRIPTION
**Pull request by satyakesav**
This pull request resolves #594.

The pull request is in progress.

The details of the pull request:
Implement Text Regressor module.

### CODE REVIEW

**Comment by satyakesav** [comment]
Tested and verified along with the performance.

File: autokeras/text/text_supervised.py [hunk 1]
```diff
@@ -174,15 +117,29 @@ def predict(self, x_test):
...
             y_preds.extend(logits)
-        return self.inverse_transform_y(y_preds)
+
+        if self.task_type:
+            return self.inverse_transform_y(y_preds)
+        else:
+            return y_preds
 
     @property
     def metric(self):
```
Comment (jhfjhfj1) [inline autokeras/text/text_supervised.py]: This if else is now written in a procedure-oriented style.
For object-oriented programming, this function should be an abstract function.
The different subclasses should override it to return different values.
The same for the loss().

File: autokeras/text/text_supervised.py [hunk 2]
```diff
@@ -36,108 +36,50 @@ class TextSupervised(SingleModelSupervised, ABC):
...
 
-        # Output directory
-        self.output_model_file = os.path.join(self.path, 'pytorch_model.bin')
+        # Output model directory
+        self.output_model_file = None
 
-        # Evaluation params
-        self.eval_batch_size = 32
+        # Task type - Regression or Classification.
+        self.task_type = None
```
Comment (jhfjhfj1) [inline autokeras/text/text_supervised.py]: I think we don't need this variable.
It should be done by different subclasses.

File: autokeras/text/pretrained_bert/modeling.py [hunk 3]
```diff
@@ -598,42 +598,29 @@ class BertForSequenceClassification(PreTrainedBertModel):
...
+        super(BertForSupervisedTasks, self).__init__(config)
         self.bert = BertModel(config)
         self.dropout = nn.Dropout(config.hidden_dropout_prob)
-        self.classifier = nn.Linear(config.hidden_size, num_labels)
+        self.classifier = nn.Linear(config.hidden_size, num_labels if num_labels else 1)
         self.apply(self.init_bert_weights)
 
+        # For regression num_labels is None.
+        if num_labels:
+            self.loss_fct = classification_loss
```
Comment (jhfjhfj1) [inline autokeras/text/pretrained_bert/modeling.py]: The loss function should be a parameter be passed in.
